### DISCUSSION TITLE
Filter on "Date Closed" for Projects Beta

### DISCUSSION THREAD

**Discussion by braymen** [community/community#17085] [Projects and Issues]
One thing that would be neat, is the ability to filter on date closed when it comes to issues. This would be neat, so we can easily grab a snapshot of the changes we did within a specific timeframe. This would be useful for those who might want to generate easy patch notes for the past two weeks.

I know there's other ways to tackle this, like just going through the commits or archiving issues after you send notes out, but a filter would make this a bit easier.

**Comment by rikkiprince** [comment]
Yeah, this would be really helpful for me when doing summaries of the past week or two of work, for management.

**Comment by nbarbettini** [comment]
Same here. I'd like to find all the issues that were closed in a date range (last month for example) for reporting purposes.

**Comment by jwerre** [comment]
I have a ton of items in my "Closed" column but they aren't an any particular order (maybe create date?). It would be great to be able to sort these by closed date.

**Comment by Ayfri** [comment]
This would be a sick feature !

**Comment by MichelML** [comment]
yesssssssssssssssssssssssssssssssssssssssssssssssssssssssssss

**Comment by ejackson** [comment]
Agreed - please!

**Comment by redcatbear** [comment]
A "closed" filter would also be really useful in the "Insights" (the automatically generated charts).

We have a lot of metrics that are time-sliced:

* Tickets solved by work category (product, support, tools, infrastructure, ...) in the last 90 days
* Lead and finish time in the last 90 days
* Trend of quality of effort estimations

The main reason why it is hard to get those directly from GitHub are missing date filters.

**Comment by bornfamousTobia** [comment]
wtDVRt6XSRFnd4810:13, 14 апреля 2023 г., "Sebastian Bär" ***@***.***>:
A "closed" filter would also be really useful in the "Insights" (the automatically generated charts).
We have a lot of metrics that are time-sliced:

Tickets solved by work category (product, support, tools, infrastructure, ...) in the last 90 days
Lead and finish time in the last 90 days
Trend of quality of effort estimations

The main reason why it is hard to get those directly from GitHub are missing date filters.

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>
-- Отправлено из мобильного приложения Яндекс Почты

**Comment by gregoirenovel** [comment]
The documentation [indicates](https://docs.github.com/en/search-github/searching-on-github/searching-issues-and-pull-requests#search-by-when-an-issue-or-pull-request-was-closed) that there is a `closed` qualifier to search issues based on the date the issues were closed.

It seems that Projects do _not_ support this qualifier at the moment though.

It would be super useful to create a view showing which tasks were completed last week, for example.

**Reply by KastanDay** [reply]
+1 to viewing "which tasks were completed last week"

**Comment by 50l3r** [comment]
I join the petition. It would be ideal to have a report of the incidents closed in the last month for example.

The closest I have found is this:

https://github.com/issues?q=is%3Aissue+closed%3A%3E%3D2023-09-01+assignee%3A50l3r+

**Comment by RichardForshaw** [comment]
I support this. Even simply to show in a project grid view what the closed dates are for issues would be useful.

**Comment by petrklapka** [comment]
This is absolutely necessary!  I'm transitioning my team from Sprints to Kanban and need:

- A way to filter by closed date so I can archive old issues from my team project and stay under the 1200 issue limit.
- See what issues my team closed during a time range without leaving the projects UX and returning to the repo issues screen

**Reply by petrklapka** [reply]
Ex:  Currently my project contains 613 closed issues and PRs.  How can I archive the ones that were closed more than 3 months ago?  (assume they are not divided up into sprints)

**Reply by tymokvo** [reply]
Under the ellipsis menu on a project, under workflows, there is an "auto-archive" workflow where you can configure this.

**Comment by otonielguajardo** [comment]
Issues have a [closed](https://docs.github.com/en/search-github/searching-on-github/searching-issues-and-pull-requests#search-by-when-an-issue-or-pull-request-was-closed) property, please make Projects support this qualifier

**Comment by tymokvo** [comment]
This is possible using the `updated` filter. E.g. `is:issue -is:open -updated:<@today-1w` yields issues where the state is not open was updated not less than 1 week ago. I.e. the timestamp on the state change is greater than the date from this time last week. A little confusing when negating inequalities, but it works.

~I think an improvement here from GitHub would be to support the `>` operator on the `updated` filter.~

The above works now. See comments for how to accomplish this.

**Reply by vmenger** [reply]
`updated` is not the same as `closed` though, for example when an issue label (or other property) is modified after the issue is closed.

**Reply by tymokvo** [reply]
I don't think this is true anymore, though it may have been in the past. I just created a project to test it. In this [view](https://github.com/users/tymokvo/projects/3/views/1) are all the issues that are closed, updated after `2025-03-19`. Issue [A](https://github.com/tymokvo/issue-test/issues/1) was closed on `2025-03-11` and tagged on `2025-03-20` and it does not show in the view using `is:closed updated:>2025-03-19`.

So, I think the query that everyone in this thread is looking for is:
```
is:closed updated:>@today-1w
```
modulo status and date offset.

**Reply by holmbergjonas** [reply]
No this is not it. Issues where fields have been updated will be included.

**Comment by paulo-hortelan** [comment]
We really need that, it's been over two years 😢

**Comment by PepaPanda** [comment]
+1 we would appreciate it. Not having this option is a blocker.

**Comment by aqu-erp** [comment]
Piling on to support this. Reducing the amount of overhead required for this kind retrospective reporting is super valuable.

**Comment by Nhan-T-Hoang** [comment]
One more vote for this. It is almost 3 years and we need this. Please !!!

**Comment by nik10-mah** [comment]
Much Needed . I need a date created and date closed columns in Github Projects mapped to created and closed dates

**Comment by tymokvo** [comment]
Hi everyone, I realize this is easy to miss because I wrote it in a comment. But I'm pretty sure that the feature everyone is requesting already exists: [see here](https://github.com/orgs/community/discussions/17085#discussioncomment-12566224).

Check out the test project/repo I made and let me know if I'm missing something.

**Comment by gerazenobi** [comment]
Little ⬆️  for this feature request form `on May 22, 2022` 😓 

Project views supporting the same `closed:2025-10-16..2025-11-05` from [issues page](https://github.com/issues).
